### ISSUE TITLE
frontend: solve problems with the button that connects the user to his wallet

### ISSUE THREAD

**Issue by armandocodecr** [Tico4Chain-Coders/dApp-Trustless-Work#69]
## Description:

The correct functioning of our button for users to connect their wallet with our dApp is extremely important. We want to avoid any kind of inconvenience and that users feel comfortable with it.
Currently the button works, but it has behaviors that prevent its fluidity.

### Problems we have detected:

1. When the user tries to connect his wallet it does not connect instantly. Sometimes he has to hit the “connect” button a second time for it to work.
2. Sometimes the wallet connects (you can tell because the “connect” button changes to “disconnect”), but even though it is connected, the sidebar does not appear. You have to refresh the page for the sidebar to appear. And this should not be like that... the user should see the sidebar and its options instantly when connecting his wallet.

The following video is a demonstration of one of the problems we have with this button: [Problem with wallet connection button | Demo video](https://www.youtube.com/watch?v=UOgYDi-UoA4)

In the video above you can see the following:

1. When I connect my wallet, the sidebar does not appear instantly. On the contrary, I have to try a second time to connect my wallet to get it to work.
2. When I click on the “disconnect” button, it does not redirect me instantly to the homepage which is in the path “/”.

**Note: The above problems do not appear in all cases. They are problems that sometimes we have detected and sometimes we have been detected by other users.**

## Requirements:

1. Solve the aforementioned problems and any others that may be contemplated.
2. The button should work in such a way that when the user connects his wallet, the sidebar options should appear instantly, otherwise it should not.
3. When the user has already connected his wallet and now wants to disconnect it, at the moment of hitting the “disconnect” button it should immediately disappear and return him to the homepage located in the root path “/”.
4. The user should not be able to enter any path other than the homepage unless he has connected his wallet.

## Validations:

1. The user must connect his wallet correctly and it must work the first time.
2. The user will be able to access the other pages when his wallet is connected. Otherwise he/she will only be able to access the homepage.
3. The user must be redirected immediately to the homepage by clicking on the “disconnect” button.

⚠️ **Please follow the guidelines for requesting an issue, launching commits, and describing PRs. Otherwise, they will be ignored.** ⚠️

**Comment by Michaelkingsdev** [comment]
Can I handle this task?

**Comment by Benjtalkshow** [comment]
I am a Full Stack Developer with a strong background in blockchain and extensive experience in Next.js/React, TypeScript, and Rust. I’ve made over 49 contributions to over 17 projects in the OnlyDust ecosystem, solving complex issues and delivering efficient, scalable solutions.

I can handle this task.
ETA is 2days.

**Comment by emarc99** [comment]
I'd like to handle this task.

**Comment by Yunusabdul38** [comment]
May I try my hand at this?

**Comment by Jagadeeshftw** [comment]
Is it okay if I take this?

**Comment by Supa-mega** [comment]
Can I take this from here?

**Comment by 1nonlypiece** [comment]
Could I try solving this?

**Comment by Birdmannn** [comment]
Mind if I take this issue?

**Comment by salazarsebas** [comment]
Hi Trustless Work,

# My background

I'm Sebastian, software engineer 👨🏻‍💻 and part of the Dojo Coding ⛩️ community, and I would like to contribute to this issue. I have experience with different technologies and I have worked on some projects through OnlyDust and I like to contribute to help the projects.

# How I plan on tackling this issue
To solve this issue, I will perform the following steps:
- Check where the button loses fluidity.
- Improve the flow to generate a better fluidity when using the button to connect your wallet.
- Test the flow in several browsers and make sure that the page does not need to be refreshed for it to work properly.
- Ask questions if needed and maintain good communication with mainteiners.
- Follow the git guide and send a pr waiting for feedback.

**Comment by respp** [comment]
Hello Trustless team!
I am Renzo Barcos, a member of the Dojo community, and in this hackathon, I had the opportunity to make my first contribution to OnlyDust with you! If possible, I would love to continue contributing through this issue.For more information about me, visit my website: www.renzobarcos.site.
My plan to resolve the issue would consist of debugging through logs and tracking the problem to identify its root causes. It could be resolved through routing, creating events to optimize response time, useEffects, among others. Once resolved effectively, and if necessary, I could implement animations to enhance the user interface experience.

**Comment by armandocodecr** [comment]
This issue is for you @salazarsebas ! 🚀
Remember to follow all the guidelines, both the ones inside the repository and the guideline to perform the PR.

Many thanks to all the other dev's who sent in their application. 🔥

**Comment by salazarsebas** [comment]
Thank you very much @armandocodecr , as soon as possible I will work on the issue. 👨🏻‍💻
I will follow all the guidelines to develop for Trustless Work, for me it's an honor. 🫡🚀

**Comment by salazarsebas** [comment]
Hi team,
Subsequently bringing changes to my fork I noticed that the wallet flow was greatly improved. I made a video doing tests in three different browsers with three different wallets and they give me as a result good fluidity. I'm glad the issue is no longer present and I hope to be able to contribute with an issue in the future. 🫡
Attached is the video of the above mentioned.


https://github.com/user-attachments/assets/5822a74d-62cc-4613-8e2b-0ed17a10d0a1

An apology for the quality, github has only 10mb for resources. ;(
